### DISCUSSION TITLE
modbus tcp integration

### DISCUSSION THREAD

**Discussion by helander** [arendst/Tasmota#16456] [Q&A]
I have a tasmota device connected to my electricity meter. I also have a solar Inverter that I can connect to using modbus tcp. I would like to read a few modbus registers on the Inverter from my tasmota device. Is that possible?

**Comment by barbudor** [comment]
The interrest of Tasmota with ModBus meters is to read ModBus Serial and make the value available over MQTT or ModBus TCP
If the meter is already connected through ModBus TCP, it not much the purpose of Tasmota and you can probably do that directly in your home automation software

**Reply by helander** [reply]
I agree that this would be the most natural way of doing things - this is what I am doing today. However being able to do what I asked for can be of interest in some cases. My question remains, is that possible?

**Reply by barbudor** [reply]
It doesn't exist today in Tasmota as a feature. 
The recently added MODBUS bridge feature can help making a serial MODBUS device appear like a TCP MODBUS one (on top of its MODBUS to mqtt capabilities) but AFAIK it does allow to talk to a TCP MODBUS device. 

Everything is possible when someone is motivated to create such feature. On ESP32 with the Berry scripting language (python like) which offer TCP features it might be reasonably easy to implement.

**Reply by helander** [reply]
Ok. Unfortunately my devices are not ESP32. Did you mean to write ”AFAIK it does NOT allow” or do you think that some support is available to talk to a TCP MODBUS device? If so can you explain a bit more?

**Reply by barbudor** [reply]
Unfortunately I meant 'not'
Current TCP Bridge allows to bridge a serial MODBUS device to TCP

**Reply by barbudor** [reply]
Check the on going discussions about MODBUS bridge

**Reply by nikito7** [reply]
"rtuovertcp"
